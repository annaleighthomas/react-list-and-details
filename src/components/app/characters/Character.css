
.Character {
  list-style-type: none;
  background: rgb(159, 246, 157);
  border: 1px solid rgb(230, 162, 162);
  padding: 10px;
  display: inline;
  grid-template-rows: 30px 120px 20px;
  grid-gap: 10px;
  justify-items: center;
  text-decoration: none;
  color: inherit;
  cursor: pointer;
  width: 10rem;

}

.Character img {
  height: 10rem;
  width: 10rem;
  object-fit: contain;
}

.Character p {
  margin: 0;
  font-size: .9rem;
  text-align: center;
}