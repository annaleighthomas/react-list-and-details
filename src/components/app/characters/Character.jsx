import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Character = ({ name, image, id }) => (
  <figure>
    <Link to={`/detailpage/${id}`}>
      <img src={image} alt={name} /></Link>
    <figcaption>
      <p>{name}</p>
    </figcaption>
  </figure >
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default Character;
