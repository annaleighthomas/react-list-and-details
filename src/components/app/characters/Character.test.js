import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('renders a Character', () => {
    const { asFragment } = render(
      <Character
        name="Text Character"
        species="Human"
        status="Alive"
        image="http://image.com/image.png"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
