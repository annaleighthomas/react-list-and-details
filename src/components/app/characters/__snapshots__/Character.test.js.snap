// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a Character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Text Character"
      src="http://image.com/image.png"
    />
    <figcaption>
      <p>
        Text Character
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
